recipes/uni-algo/all/conanfile.py

```diff
@@ -69,24 +67,26 @@ def package_id(self):
             self.info.clear()
 
     def validate(self):
-        if not is_msvc(self):
-            minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
-            if minimum_version and Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration(
-                    f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
-                )
-        else:
-            check_min_vs(self, int(self._compilers_minimum_version["msvc"]))
-            if self.options.get_safe("shared"):
-                raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared on Visual Studio and msvc.")
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, self._min_cppstd)
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+        if is_msvc(self) and self.options.get_safe("shared"):
+            raise ConanInvalidConfiguration(f"{self.ref} can not be built as shared with msvc")
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
 
     def generate(self):
-        tc = CMakeToolchain(self)
-        tc.variables["UNI_ALGO_HEADER_ONLY"] = self.options.header_only
```

Comment:
I prefer when this is set for all cases and the value is changed, but it's in no way a blocker for me

Reply:
I prefer to not keep a logic which doesn't make sense. Here conan_toolchain.cmake is not used at all if header only in current recipe logic. So keeping this logic would confuse maintainers, since it may mean that it's mandatory when header only is True.